src/signature.c

```diff
@@ -230,12 +231,20 @@ gboolean cms_verify(GBytes *content, GBytes *sig, GError **error) {
 		goto out;
 	}
 
+	if (cms)
+		*cms = icms;
+
+	if (store)
+		*store = istore;
+
+
 	res = TRUE;
 out:
 	ERR_print_errors_fp(stdout);
 	BIO_free_all(incontent);
 	BIO_free_all(insig);
-	X509_STORE_free(store);
+	if (!store)
+		X509_STORE_free(istore);
```

Comment:
Do we need to free icms as well if we don't pass it along?

Reply:
We did not free former `cms` variable, yet. Added a separate fix commit for this and adapted further patches hande `icms` properly.